app/services/schools/generate_chart_configuration.rb

```diff
@@ -0,0 +1,57 @@
+require 'dashboard'
+
+module Schools
+  class GenerateChartConfiguration
+    def initialize(
+        school,
+        aggregated_meter_collection,
+        fuel_types_for_analysis = school.fuel_types_for_analysis,
+        dashboard_fuel_types = DashboardConfiguration::DASHBOARD_FUEL_TYPES,
+        dashboard_page_configuration = DashboardConfiguration::DASHBOARD_PAGE_GROUPS
+      )
+
+      @school = school
+      @aggregated_meter_collection = aggregated_meter_collection
+      @fuel_types_for_analysis = fuel_types_for_analysis
+      @dashboard_fuel_types = dashboard_fuel_types
+      @dashboard_page_configuration = dashboard_page_configuration
+    end
+
+    def generate
+      Rails.logger.info "Generating chart configuration for #{@school.name} - using default values"
+      page_and_chart_config = {}
+      pages.each do |page|
+        page_and_chart_config[page.to_sym] = page_config(page)
+      end
+      @school.configuration.update(dashboard_charts: page_and_chart_config)
+    end
+
+  private
+
+    def _remove?(chart_type, chart_config = { y_axis_units: :kwh })
+      output = ChartData.new(@aggregated_meter_collection, chart_type, false, chart_config).data
+      output.first.series_data.nil?
+    rescue EnergySparksNotEnoughDataException
```

Comment:
Need more exceptions here. Possibly all the ones from the equivalences?

Reply:
Yes, I'll review that properly when we implement this method - made a note in Trello

Reply:
Gives us some confidence using kwh, but in the future we might want to ensure a chart works properly, if it is a scalable one, against all possible units.